extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/AuthenticationFailedExceptionMapper.java

```diff
@@ -37,18 +37,25 @@ public Response toResponse(AuthenticationFailedException exception) {
             if (authenticator != null) {
                 ChallengeData challengeData = authenticator.getChallenge(context)
                         .await().indefinitely();
-                Response.ResponseBuilder status = Response.status(challengeData.status);
-                if (challengeData.headerName != null) {
-                    status.header(challengeData.headerName.toString(), challengeData.headerContent);
+                int statusCode = challengeData == null ? 401 : challengeData.status;
+                Response.ResponseBuilder responseBuilder = Response.status(statusCode);
+                if (challengeData != null && challengeData.headerName != null) {
+                    responseBuilder.header(challengeData.headerName.toString(), challengeData.headerContent);
+                }
+                if (LaunchMode.isDev() && exception.getMessage() != null) {
```

Comment:
I believe we should only do it for `401`, where returning an exception message in the devmode can help users.
If it is 302, the user won't see anything, for OIDC, 302 is a (re-)authentication for whatever reasons, so IMHO we could avoid returning text with 302 .

Reply:
I am not sure why you don't want to show it in case someone doesn't follow redirect, but I trust you know what you are doing. There is too much exceptions and wrapping for me to determine if it can really be useful to show body in any 302 scenario at all. I'll change it.